### ISSUE TITLE
point event on heatmap not work well were the array contain more 5000 points 

### ISSUE THREAD

**Issue by neriaaa46** [highcharts/highcharts#17200]
I use vue.js framework
I want do display on heatmap three different data  with different color and with different range of colors of all data. 
I use this code to display the data : 
in the html:  
        <div class="highcharts-container">
          <highcharts :options="options" class="graph" ref="highcharts"></highcharts>
        </div>

the data: 

options: {
        chart: {
          type: 'heatmap',
          zoomType: 'xy',
          marginTop: 60,
          marginBottom: 60,
          plotBorderWidth: 1,
          height: 450,
          width: 520,
        },
        title: {
          text: 'Correlated Mutations'
        },
        subtitle: {
          text: 'Source: Heinz  2003'
        },
        colorAxis: [
          {stops: [
            [0, '#dafaff'],
            [0.25, '#adf2ff'],
            [0.5, '#73e0ec'],
            [0.75, '#28c7d9'],
            [1, '#01899f']
          ],
          min: 0,
          max: 1,
          startOnTick: true,
          endOnTick: true },
        {stops: [
            [0, '#efd4ff'],
            [0.25, '#d89eff'],
            [0.5, '#c36cff'],
            [0.75, '#ae2eff'],
            [1, '#8a00e3']
          ],
          min: 0,
          max: 1,
          startOnTick: true,
          endOnTick: true },
        {stops: [
            [0, '#ffd1d1'],
            [0.25, '#ffa9a9'],
            [0.5, '#fd6767'],
            [0.75, '#c70000'],
            [1, '#8c0000']
          ],
          min: 0,
          max: 1,
          startOnTick: true,
          endOnTick: true }
        ],
        xAxis: {
          title: {
            enabled: true,
            text: 'Height (cm)'
          },
        },
        yAxis: {
          title: {
            text: 'Weight (kg)'
          }
        },
        legend: {
          align: 'right',
          layout: 'vertical',
          marginLeft: 30,
          verticalAlign: 'top',
          y: 40,
          symbolHeight: 105,
        },
        plotOptions: {
          scatter: {
            marker: {
              radius: 5,
              states: {
                hover: {
                  enabled: true,
                  lineColor: 'rgb(100,100,100)'
                }
              }
            },
            states: {
              hover: {
                marker: {
                  enabled: false
                }
              }
            },
            tooltip: {
              headerFormat: '<b>{series.name}</b><br>',
              pointFormat: '{point.x} , {point.y}'
            }
          },
        },
        series: [{
          name: 'Structure 1',
          colorAxis: 0,
          visible: true,
          point: {
            events: {
              click: (e) => {
                this.pointClick(e)
              }
            }
          },
          data: [[100, 100, 0.294025, 0.9], [100, 101, 0.394025, 0.1],[100, 102, 0.194025, 0.9], [99, 103, 0.194025, 0.9]]
        }, {
          name: 'Structure 2',
          colorAxis: 1,
          visible: true,
          point: {
            events: {
              click: (e) => {
                this.pointClick(e)
              }
            }
          },
          data: [[104, 100, 0.294025, 0.9], [100, 105, 0.394025, 0.1],[100, 106, 0.194025, 0.9], [99, 107, 0.194025, 0.9]]
        }, {
            name: 'Structure 3',
            colorAxis: 2,
            visible: true,
            point: {
              events: {
                click: (e) => {
                  this.pointClick(e)
                }
              }
            },
            data: [[90, 90, 0.294025, 0.9], [110, 110, 0.394025, 0.1],[100, 102, 0.194025, 0.9], [99, 111, 0.194025, 0.9],
              [90, 108, 0.964025, 0.9], [99, 109, 0.394025, 0.1],[99, 110, 0.994025, 0.9], [98, 111, 0.274025, 0.9]]
          }
        ]
      }
    }

and the event of point work well I can use the data of the point that I clicked , but 
when I use fetch to get data and the options.series[i].data get array with more 4999 points the event not work well just 
to the this series[i]. 

I tried to add event in this options: 
 document.querySelector('.highcharts-container').addEventListener('click', e => {
            console.log({x: Math.round(e.xAxis[0].value), y: Math.round(e.yAxis[1].value)})
 });

and it work if there is more 4999 points in series[i], but i dont no which data the the event came 
series[0].data or series[1].data or series[2].data ? 
It is important which from which data the event came from. 

thank for your help :)

**Comment by pawellysy** [comment]
Hello, @neriaaa46 thanks for creating this issue.
I am having trouble reproducing your problem. Could you create a live demo using this link: https://jsfiddle.net/BlackLabel/9fd37vob/

Thanks, and sorry for the inconvenience!

**Comment by neriaaa46** [comment]
Hi, i will create demo and send to you.
Can i saved the changes on the link and sand back ?
Thank !






בתאריך יום חמישי, 21 באפריל 2022, מאת pawellysy ***@***.***>:

**Comment by neriaaa46** [comment]
Hi, this is the link of demo what i try to do
https://jsfiddle.net/sj37nbe9/39/,
I use here with function to create random array of data, in my real project
i fetch data from server

I explain what I'm trying to do:
I use a heatmap with three different data and colors, when I click on a
point in the heatmap, I want to get the data point and use this data point.
array of data looks like this - [[5,17,0.5],[14,57,0.5],..............]
In my project when the length of the array is longer than 4999 the event to
get points not working i cant get the data point.
In this demo link it works well. The length of the array is longer than
4999,  but in my production code it's not working well and I
don't understand why.

I remind,  I work with vue.js on the client side.

I hope you can look at the demo and understand what I am trying  to do,
and then suggest an idea to help to solve this issue.
Thank you very much !!














‫בתאריך יום ה׳, 21 באפר׳ 2022 ב-18:01 מאת ‪pawellysy‬‏ <‪
***@***.***‬‏>:‬

**Comment by neriaaa46** [comment]
I Found the problem, you can close this issue. :)

**Comment by pawellysy** [comment]
Ok, could you share what the problem was so that we could know in the future? Thanks!

**Comment by neriaaa46** [comment]
I search vue.js Highcharts example. 
I found example and I change it to my demo to check if it works. 
The demo was worked well and I tried to figure out what the different from my production code. 
I found that in the demo there is import I didnt use -> import hcHeatmap from "highcharts/modules/heatmap" 
and after this import there is loading to the highcharts ->  hcHeatmap(Highcharts)

So it was different from my code I change it and the click event was work. 
if you want to look at my vue demo the - link: https://codesandbox.io/s/vue-template-forked-niig7v?file=/src/components/Chart.vue:234-257.

Thank for your help !!!
